addons/sourcemod/scripting/ttt/ttt_inventory.sp

```diff
@@ -0,0 +1,690 @@
+#include <ttt>
+#include <ttt_shop>
+#include <ttt_inventory>
+
+#define PLUGIN_NAME TTT_PLUGIN_NAME ... " - Inventory"
+
+enum eInventory
+{
+	userid,
+	availableCredits,
+	StringMap:hInventoryItems
+}
+
+enum eLootChance
+{
+	Float:lootchance_credit,
+	Float:lootchance_item
+}
+
+bool g_bCreditLoot;
+bool g_bItemLoot;
+
+char g_szPluginTag[64];
+
+float g_fCreditLootChance;
+float g_fItemLootChance;
+float g_fLootChanceFallOff;
+
+int g_iCreditLootMax;
+int g_iItemLootMax;
+
+ConVar g_cv_PluginTag = null;
+ConVar g_cv_EnableCreditLoot = null;
+ConVar g_cv_EnableItemLoot = null;
+ConVar g_cv_CreditLootMax = null;
+ConVar g_cv_ItemLootMax = null;
+ConVar g_cv_CreditLootChance = null;
+ConVar g_cv_ItemLootChance = null;
+ConVar g_cv_LootChanceFallOff = null;
+
+Handle g_hOnInventoryReady = null;
+Handle g_hOnInventoryMenuItemSelect = null;
+
+StringMap g_PlayerInventory = null;
+StringMap g_PlayerLootChance = null;
+
+public Plugin myinfo =
+{
+    name = PLUGIN_NAME,
+    author = TTT_PLUGIN_AUTHOR,
+    description = TTT_PLUGIN_DESCRIPTION,
+    version = TTT_PLUGIN_VERSION,
+    url = TTT_PLUGIN_URL
+};
+
+public APLRes AskPluginLoad2(Handle myself, bool late, char[] error, int err_max)
+{
+	g_PlayerInventory = CreateTrie();
+	
+	g_hOnInventoryReady = CreateGlobalForward("TTT_OnInventoryReady", ET_Ignore);
+	g_hOnInventoryMenuItemSelect = CreateGlobalForward("TTT_OnInventoryMenuItemSelect", ET_Ignore, Param_Cell, Param_String);
+	
+	CreateNative("TTT_AddInventoryItem", Native_AddInventoryItem);
+	CreateNative("TTT_RemoveInventoryItem", Native_RemoveInventoryItem);
+	CreateNative("TTT_GetInventoryListing", Native_GetInventoryListing);
+	CreateNative("TTT_IsItemInInventory", Native_IsItemInInventory);
+	
+	RegPluginLibrary("ttt_inventory");
+}
+
+public void OnPluginStart()
+{
+	TTT_IsGameCSGO();
+	LoadTranslations("ttt.phrases");
+	
+	g_cv_EnableCreditLoot = CreateConVar("ttt_enable_loot_credit", "1", "Enables players to loot credits from dead bodies.");
+	g_cv_EnableItemLoot = CreateConVar("ttt_enable_loot_item", "1", "Enables players to loot items from dead bodies.");
+	g_cv_CreditLootMax = CreateConVar("ttt_loot_credit_max", "0", "The amount of credits a player can loot. 0 for unlimited.");
+	g_cv_ItemLootMax = CreateConVar("ttt_loot_item_max", "0", "The amount of items a player can loot per dead body. Duplicates count as an item. 0 for unlimited.");	
+	g_cv_CreditLootChance = CreateConVar("ttt_loot_credit_chance", "0.2", "Chance to loot credits. Default: 0.2 meaning 20%.");
+	g_cv_ItemLootChance = CreateConVar("ttt_loot_item_chance", "0.3", "Chance to loot items. Default: 0.3 meaning 30%.");
+	g_cv_LootChanceFallOff = CreateConVar("ttt_loot_chance_falloff", "0.5", 
+	"Upon successful looting, the ratio at which your next loot chance is reduced by. Default: 0.5 meaning loot chance is reduced by 50%.");
+	
+	RegConsoleCmd("sm_ttt_inventory", Command_Inventory);
+	
+	HookConVarChange(g_cv_EnableCreditLoot, OnConvarChanged);
```

Comment:
You don't need anymore. I prefer to use methodmaps, cvars are already cached you don't need for each cvar a new variable. (Plugin Tag is fine)

Reply:
Somewhat of a habit since it is faster during runtime to have the cvar cached in the plugin. Can use the methodmap version sure. Get back to me on this one.